### PR TITLE
[RHELC-780] Improve logging and handling of RHSM facts

### PR DESCRIPTION
**Pull request by r0x0d**
After merging the #612, we left behind a few comments to be addressed in a future PR of some things that we could improve.

Signed-off-by: Rodolfo Olivieri <rolivier@redhat.com>

Jira Issue: [RHELC-780](https://issues.redhat.com/browse/RHELC-780)

Checklist
- [ ] PR meets acceptance criteria specified in the Jira issue
- [x] PR has been tested manually in a VM (either author or reviewer)
- [x] Jira issue has been made public if possible
- [x] `[RHELC-]` is part of the PR title 
- [ ] Code and tests are documented properly
- [x] The commits are squashed to as few commits as possible (without losing data) 
- [ ] When merged: Jira issue has been updated to `Release Pending`

### CODE REVIEW

**Comment by kokesak** [comment]
(just trying if the packit rpm build works)

File: convert2rhel/unit_tests/breadcrumbs_test.py [hunk 1]
```diff
@@ -65,7 +65,11 @@ def test_finish_collection(pretend_os, success, monkeypatch):
     save_migration_results_mock = mock.Mock()
     save_rhsm_facts_mock = mock.Mock()
 
-    monkeypatch.setattr(breadcrumbs.breadcrumbs, "_save_migration_results", save_migration_results_mock)
+    monkeypatch.setattr(
+        breadcrumbs.breadcrumbs,
+        "_save_migration_results",
+        save_migration_results_mock,
```
Comment (abadger) [inline convert2rhel/unit_tests/breadcrumbs_test.py:71]:
```suggestion
        save_migration_results_mock
```

File: convert2rhel/breadcrumbs.py [hunk 2]
```diff
@@ -172,16 +177,20 @@ def _save_migration_results(self):
 
     def _save_rhsm_facts(self):
         """Write the results of the breadcrumbs to the rhsm custom facts file."""
+        if not os.path.exists(RHSM_CUSTOM_FACTS_FOLDER):
+            loggerinst.debug("No RHSM facts folder found at '%s'. Creating a new one..." % RHSM_CUSTOM_FACTS_FOLDER)
+            # Using mkdir_p here as the `/etc/rhsm` might not exist at all.
+            # Usually this can happen if we fail in the first run and we want to
+            # save the custom facts gathered so far, or, if the --norhsm option
+            # is provided.
```
Comment (abadger) [inline convert2rhel/breadcrumbs.py:185]:
```suggestion
            # is provided.
            # This is safe as the RHSM_CUSTOM_FACTS_FOLDER, /etc/rhsm/facts, and its parents
            # are only writable by root
```

**Comment by abadger** [comment]
I think this comment from the jira ticket is not yet addressed by this PR: https://github.com/oamg/convert2rhel/pull/612#issuecomment-1294199823

**Comment by r0x0d** [comment]
> I think this comment from the jira ticket is not yet addressed by this PR: [#612 (comment)](https://github.com/oamg/convert2rhel/pull/612#issuecomment-1294199823)

That was fixed as part of another PR, @abadger. This one https://github.com/oamg/convert2rhel/pull/660

**Review by abadger — APPROVED** [review]
Everything looks good here.  Approved.

**Pull request merged**